### PR TITLE
Added petting as a scriptable component.

### LINKED ISSUE #2969: Cannot pet ian
Description
You cannot currently pet ian.

Steps to Reproduce:  

1.  switch to help intent
2.  click ian
3.  cry

### PR DESCRIPTION
**Pull request by h3killa**
# Pull Request Template

### Purpose
Added capability to pet NPC's via Script component. 
Fixes #2969 

### Notes:
Added as script component to be applied to any relevant Prefab. Added to NPC_Ian Prefab for testing.

### Please make sure you have followed the self checks below before submitting a PR:

- [x] - Code is sufficiently commented
- [x] - Code is indented with tabs and not spaces
- [x] - The PR does not include any unnecessary .meta, .prefab or <b>.unity (scene) changes</b>
- [x] - The PR does not bring up any new compile errors
- [x] - The PR has been tested in editor
- [x] - Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [x] - Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- [x] - Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- [ ] - The PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [x] - The PR has been tested with round restarts.
- [ ] - The PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

### CODE REVIEW

**Comment by Misterixmax** [comment]
Maybe add this also for Runtime?

**Comment by DooblyNoobly** [comment]
Missing the meta file for the new Pettable script. Can you include that so I can review again.

File: UnityProject/Assets/Scripts/Player/Pettable.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	public bool Interact(PositionalHandApply interaction)
+	{
+		var targetNPCHealth = interaction.TargetObject.GetComponent<LivingHealthBehaviour>();
+		var performerPlayerHealth = interaction.Performer.GetComponent<PlayerHealth>();
+		var performerRegisterPlayer = interaction.Performer.GetComponent<RegisterPlayer>();
+
+		// Is the target in range for a pet? Is the target conscious for the pet? Is the performer's intent set to help?
+		// Is the performer's hand empty? Is the performer not stunned/downed? Is the performer conscious to perform the interaction?
+		// Is the performer interacting with itself?
+		if (!PlayerManager.LocalPlayerScript.IsInReach(interaction.WorldPositionTarget, false) ||
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Player/Pettable.cs]: Why not use `Validations.CanApply` instead? it has all default validations in place

**Review by fomalsd — COMMENTED** [review]
It's fine, but own set of validations is unnecessary here, just use `Validations.CanApply` instead

File: UnityProject/Assets/Scripts/Player/Pettable.cs (hunk 1 above)
Comment (h3killa) [reply UnityProject/Assets/Scripts/Player/Pettable.cs]: Its basically a copy paste of huggable.cs. Went ahead and converted it over to `Validations.CanApply`.

Comment (h3killa) [reply UnityProject/Assets/Scripts/Player/Pettable.cs]: Not 100% regarding the NetworkSide so if that needs adjusted just give me a heads up.

File: UnityProject/Assets/Scripts/Player/Pettable.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using UnityEngine;
+
+/// <summary>
+/// Allows an object to be pet by a player. Shameless copy of Huggable.cs
+/// </summary>
+public class Pettable : MonoBehaviour, IClientInteractable<PositionalHandApply>
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Player/Pettable.cs]:
```suggestion
public class Pettable : MonoBehaviour, ICheckedInteractable<PositionalHandApply>
```

File: UnityProject/Assets/Scripts/Player/Pettable.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{
+	public bool Interact(PositionalHandApply interaction)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Player/Pettable.cs]:
```suggestion
	public bool WillInteract( PositionalHandApply interaction, NetworkSide side )
```

File: UnityProject/Assets/Scripts/Player/Pettable.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		var targetNPCHealth = interaction.TargetObject.GetComponent<LivingHealthBehaviour>();
+		var performerPlayerHealth = interaction.Performer.GetComponent<PlayerHealth>();
+		var performerRegisterPlayer = interaction.Performer.GetComponent<RegisterPlayer>();
+
+		// Is the target in range for a pet? Is the target conscious for the pet? Is the performer's intent set to help?
+		// Is the performer's hand empty? Is the performer not stunned/downed? Is the performer conscious to perform the interaction?
+		// Is the performer interacting with itself?
+		if (!Validations.CanApply(interaction.Performer, interaction.TargetObject, NetworkSide.Server, true, ReachRange.Standard, interaction.TargetVector))
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Player/Pettable.cs]:
```suggestion
		if (!Validations.CanApply(interaction.Performer, interaction.TargetObject, side, true, ReachRange.Standard, interaction.TargetVector))
```

**Review by fomalsd — COMMENTED** [review]
Oh, didn't notice it's not a checked interactable. Here are suggested fixes. Otherwise this would only be a clientside interaction and server won't ever know about it.
see https://unitystation.github.io/unitystation/development/Implementing-Interactions/#why-should-i-implement-icheckedinteractable-instead-of-just-iinteractable

**Comment by DooblyNoobly** [comment]
Checked the PR out on a clean git and got this error:

image

**Comment by fomalsd** [comment]
oh indeed, you're supposed to move whatever's going to happen to the ServerPerformInteraction method, this part in particular:
```
		Chat.AddActionMsgToChat(interaction.Performer,
	$"You pet {interaction.TargetObject.name}.", $"{interaction.Performer.ExpensiveName()} pets {interaction.TargetObject.name}.");
```

**Pull request merged**